pkg/fileutil/util_test.go

```diff
@@ -75,27 +95,30 @@ func TestLoadDynamicFile(t *testing.T) {
 		testA.mutex.Unlock()
 	}
 	for _, c := range cases {
-		updateFile(testA, c.input, t)
-		testA.mutex.Lock()
-		if testA.expectedContent != c.want {
-			t.Errorf(
-				"Unexpected result: TestLoadDynamicFile: got: %s, wanted %s",
-				testA.expectedContent,
-				c.want,
-			)
-		}
-		testA.mutex.Unlock()
+		updateFile(f.Name(), c.input, t)
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		go wait.Until(func() {
+			testA.mutex.Lock()
```

Comment:
what does the goroutine do, won't it race with the next test case? like a) update abcd; b) poll every 100ms for 1s for DyamicFile to read abc;  c) update fgh: won't b and c overlap?

(since the test passes I think I'm misunderstanding something)

Reply:
or maybe it's passing because the goroutine gets canceled immediately by cancel()